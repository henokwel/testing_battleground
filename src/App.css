.App {
  background-color: #282c34;
  min-height: 100vh;
  color: whitesmoke;
}


.item{
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  max-width: 250px;
  background-color: black;
  margin: 10px 0px 10px 5px;
  padding: 5px;
}
.item_btn_Container{
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 30%;
  min-height: 50px;
}