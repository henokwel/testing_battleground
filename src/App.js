import './App.css';
import { ItemList } from './components/ItemList'
import { useContext, useState } from 'react';
import { Context } from './context/MyContext';






function App() {
  const [currentTitle, setCurrentTitle] = useState("")

  const { dispatch, state } = useContext(Context)

  const handleEdit_Title = () => {
    // if current is false, toggle
    if (currentTitle.trim() === "") return

    dispatch({
      type: "add",
      payload: { id: `${Math.floor(Math.random() * 300)}`, title: currentTitle, done: false }
    })

    setCurrentTitle("")
  }



  const handleChange = (e) => {
    const target = e.target
    const value = target.value
    console.log(value);

    // if (!value || value === title) return
    setCurrentTitle(value)
  }


  return (
    <div className="App">
      <header onSubmit={e => e.preventDefault()}>
        <input type="text" value={currentTitle} placeholder='Hello' onChange={e => handleChange(e)} />
        <input type="submit" title='Submit' onClick={handleEdit_Title} />
      </header>

      <main>
        <div>
         </div>
        <ItemList data={[]} />
      </main>
    </div>
  );
}

export default App;
