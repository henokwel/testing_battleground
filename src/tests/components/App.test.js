import { shallow } from "enzyme";
import App from "../../App";
import { MyProvider } from "../fixtures/provider";



describe('App', () => {
    test('should render App', () => {

        const wrapper = shallow(<App />,
            { wrappingComponent: MyProvider }
        )
        expect(wrapper).toMatchSnapshot()
    })

})
