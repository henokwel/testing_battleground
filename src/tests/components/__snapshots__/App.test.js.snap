// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App 1`] = `
<div
  className="App"
>
  <header
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Hello"
      type="text"
      value=""
    />
    <input
      onClick={[Function]}
      title="Submit"
      type="submit"
    />
  </header>
  <main>
    <div />
    <ItemList />
  </main>
</div>
`;
